### ISSUE TITLE
Support let-else return statements

### ISSUE THREAD

**Issue by SOF3** [taiki-e/auto_enums#152]
```rs
error: `#[auto_enum]` is required two or more branches or marker macros in total, there is only one branch or marker macro in this statement
 --> src/lib.rs:4:1
  |
4 | / pub fn x(input: Option<String>) -> impl Iterator<Item = u8> {
5 | |     let Some(s) = input else {
6 | |         return marker!(std::iter::empty());
7 | |     };
8 | |     marker!(s.into_bytes().into_iter())
9 | | }
  | |_^
```

**Comment by taiki-e** [comment]
This is because syn 1.0 parses let-else as Expr::Verbatim, so this particular case should be automatically supported if we update to syn 2.0.

**Comment by taiki-e** [comment]
Should be resolved by #158.
